contributingGuides/review/RULES.md

```diff
@@ -0,0 +1,203 @@
+# ✅ AI Code Review Rules
+
+These rules are used to conduct structured code reviews on pull request diffs. Each rule includes:
+
+- A unique **Rule ID**
+- **Severity level**: MUST / SHOULD / OPTIONAL
+- **Pass/Fail condition**
+- Examples of good and bad usage
+
+---
+
+## Performance Rules
+
+### [PERF-1] No spread in list item's renderItem
+- **Severity**: MUST
+- **Condition**: When passing data to components in renderItem functions, avoid using spread operators to extend objects. Instead, pass the base object and additional properties as separate props to prevent unnecessary object creation on each render.
+
+Good:
+```tsx
+<Component
+  item={item}
+  isSelected={isSelected}
+  shouldAnimateInHighlight={isItemHighlighted}
+/>
+```
+
+Bad:
+```tsx
+<Component
+  item={{
+      shouldAnimateInHighlight: isItemHighlighted,
+      isSelected: selected,
+      ...item,
+  }}
+/>
+```
+
+---
+
+### [PERF-2] Use Set for O(1) lookups instead of array methods
+- **Severity**: SHOULD
+- **Condition**: When performing repeated lookups or checking existence in collections, convert arrays to Sets for O(1) performance instead of using O(n) methods like `.includes()`.
+
+Good:
+```tsx
+const reportIdFromRoute = report.reportId;
+const reportUserIdSet = new Set(Object.values(reports));
+const reportExists = reportUserIdSet.has(reportIdFromRoute);
+```
+
+Bad:
+```tsx
+const reportIdFromRoute = report.reportId;
+const reportExists = Object.values(reports).includes(reportIdFromRoute);
+```
+
+---
+
+### [PERF-3] Use early returns in array iteration methods
```

Comment:
Why `in array iteration methods`? I think early return and early throw is a general best practice, in fact it's in our style guides already